mne/viz/utils.py

```diff
@@ -2242,3 +2250,192 @@ def _grad_pair_pick_and_name(info, picks):
+        if len(nan_names) > 0:
+            nan_picks = pick_channels(info['ch_names'], nan_names)
+            projector[nan_picks] = np.nan
+    elif proj:
+        projector, _ = setup_proj(info, add_eeg_ref=False, verbose=False)
+    return projector, whitened_ch_names
+
+
+def _set_ax_label_style(ax, params, italicize=True):
+    import matplotlib.text
+    for tick in params['ax'].get_yaxis().get_major_ticks():
+        for text in tick.get_children():
+            if isinstance(text, matplotlib.text.Text):
+                whitened = text.get_text() in params['whitened_ch_names']
+                whitened = whitened and italicize
+                text.set_style('italic' if whitened else 'normal')
+
+
+def _check_sss(info):
+    """Check SSS history in info."""
+    ch_used = [ch for ch in ['eeg', 'grad', 'mag']
+               if _contains_ch_type(info, ch)]
+    has_meg = 'mag' in ch_used and 'grad' in ch_used
+    has_sss = (has_meg and len(info['proc_history']) > 0 and
+               info['proc_history'][0].get('max_info') is not None)
+    return ch_used, has_meg, has_sss
+
+
+def _triage_rank_sss(info, covs, rank=None, scalings=None):
+    from ..cov import _estimate_rank_meeg_cov
+    if rank is None:
+        rank = dict()
+    if scalings is None:
+        scalings = dict(mag=1e12, grad=1e11, eeg=1e5)
+
+    # Only look at good channels
+    picks = pick_types(info, meg=True, eeg=True, ref_meg=False, exclude='bads')
+    info = pick_info(info, picks)
+    ch_used, has_meg, has_sss = _check_sss(info)
+    if has_sss:
+        if 'mag' in rank or 'grad' in rank:
+            raise ValueError('When using SSS, pass "meg" to set the rank '
+                             '(separate rank values for "mag" or "grad" are '
+                             'meaningless).')
+    elif 'meg' in rank:
+        raise ValueError('When not using SSS, pass separate rank values '
+                         'for "mag" and "grad" (do not use "meg").')
+
+    picks_list = _picks_by_type(info, meg_combined=has_sss)
+    if has_sss:
+        # reduce ch_used to combined mag grad
+        ch_used = list(zip(*picks_list))[0]
+    # order pick list by ch_used (required for compat with plot_evoked)
+    picks_list = [x for x, y in sorted(zip(picks_list, ch_used))]
+    n_ch_used = len(ch_used)
+
+    # make sure we use the same rank estimates for GFP and whitening
+
+    picks_list2 = [k for k in picks_list]
+    # add meg picks if needed.
+    if has_meg:
+        # append ("meg", picks_meg)
+        picks_list2 += _picks_by_type(info, meg_combined=True)
+
+    rank_list = []  # rank dict for each cov
+    for cov in covs:
+        # We need to add the covariance projectors, compute the projector,
+        # and apply it, just like we will do in prepare_noise_cov, otherwise
+        # we risk the rank estimates being incorrect (i.e., if the projectors
+        # do not match).
+        info_proj = info.copy()
+        info_proj['projs'] += cov['projs']
```

Comment:
This is one bugfix -- the estimated rank needs to use the union of projectors to mirror what happens in computing the whitener